opsmngr/version_manifest.go

```diff
@@ -0,0 +1,105 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opsmngr
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const (
+	versionManifestBasePath   = "versionManifest"
+	versionManifestStaticPath = "https://opsmanager.mongodb.com/static/version_manifest/%s"
+)
+
+// VersionManifestService is an interface for interfacing with the Version Manifest
+// endpoints of the MongoDB Ops Manager API.
+//
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/version-manifest/
+type VersionManifestService interface {
+	Get(context.Context, string) (*VersionManifest, *atlas.Response, error)
+	Update(context.Context, *VersionManifest) (*VersionManifest, *atlas.Response, error)
+}
+
+// VersionManifestServiceOp provides an implementation of the VersionManifestService interface
+type VersionManifestServiceOp service
+
+var _ VersionManifestService = &VersionManifestServiceOp{}
+
+type VersionManifest struct {
+	Updated  int64      `json:"updated,omitempty"`
+	Versions []*Version `json:"versions,omitempty"`
+}
+
+type Version struct {
+	Name   string   `json:"name,omitempty"`
+	Builds []*Build `json:"builds,omitempty"`
+}
+
+type Build struct {
+	Architecture       string    `json:"architecture,omitempty"`
+	GitVersion         string    `json:"gitVersion,omitempty"`
+	Platform           string    `json:"platform,omitempty"`
+	URL                string    `json:"url,omitempty"`
+	MaxOSVersion       string    `json:"maxOsVersion,omitempty"`
+	MinOSVersion       string    `json:"minOsVersion,omitempty"`
+	Win2008plus        bool      `json:"win2008plus,omitempty"`
+	WinVCRedistDLL     string    `json:"winVCRedistDll,omitempty"`
+	WinVCRedistOptions []*string `json:"winVCRedistOptions,omitempty"`
+	WinVCRedistURL     string    `json:"winVCRedistUrl,omitempty"`
+	WinVCRedistVersion string    `json:"winVCRedistVersion,omitempty"`
+	Flavor             string    `json:"flavor,omitempty"`
+}
+
+// Get gets the version manifest
+//
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/version-manifest/view-version-manifest/
+func (s *VersionManifestServiceOp) Get(ctx context.Context, version string) (*VersionManifest, *atlas.Response, error) {
+	if version == "" {
+		return nil, nil, atlas.NewArgError("version", "must be set")
+	}
+
+	path := fmt.Sprintf(versionManifestStaticPath, version)
+	req, err := http.NewRequest(http.MethodGet, path, nil)
+	if err != nil {
```

Comment:
This api call doesn’t need autentication, this is why I am directly calling http.NewRequest

https://docs.opsmanager.mongodb.com/current/reference/api/version-manifest/view-version-manifest/

Reply:
I didn’t notice it 😢 , do you want me to change it? I will wait for your full review before pushing something here

Reply:
We should fix this sooner than later tho your implementation of teh base path may need to be handle in the cli

Reply:
I am going to open a PR soon

Reply:
@gssbzn I am trying to use the normal flow but the endpoint here is `https://opsmanager.mongodb.com/static/version_manifest/%s`and I think that I wouldn’t be able to use this path if I use `NewRequest`. Am I wrong?

Reply:
@andreaangiolillo you can, the `go-client-mongodb-ops-manager` never handles authentication that's independent from it, the only limitation here is that the library also doesn't handle a base path so `https://opsmanager.mongodb.com/` can't be part of your URL and people using the library will need to define that (ie mongocli will need to handle setting the base path)

Reply:
you can see how the cli handles unauthenticated clients here
https://github.com/mongodb/mongocli/blob/cab240a8c04b9db1fd049ad920ce9198f63be5c4/internal/store/store.go#L175-L202